File: bus/BusViewModel.m
//
//  BusViewModel.m
//  bus
//
//  Created by liuhongnian on 2018/11/7.
//  Copyright © 2018年 liuhongnian. All rights reserved.
//

#import "BusViewModel.h"
#import "BusService.h"

@implementation BusViewModel

@end


//    _busService = [[BusService alloc] init];
//    [_busService queryBusWithRoadName:@"81" direction:direction busStation:stationPoint completion:^(NSDictionary * _Nullable busInfo, NSError * _Nullable error) {
//
//        if (error) {
//            weakSelf.error = error;
//        }else{
//            weakSelf.result = busInfo;
//        }
//    }];


@interface Bus11ViewModel()

@property (nonatomic,strong) BusService *busService;

@end
@implementation Bus11ViewModel
- (void)startRequest11RoadWithDirection:(NSString *)direction stationName:(NSString *)stationPoint {

    __weak typeof(self) weakSelf = self;    
    _busService = [[BusService alloc] init];
    
    [_busService queryBusWithRoadName:@"11"
                            direction:direction busStation:stationPoint completion:^(NSDictionary * _Nullable busInfo, NSError * _Nullable error) {
                                
                                if (error) {
                                    weakSelf.error = error;
                                }else{
                                    weakSelf.result = busInfo;
                                }
        
    }];
    
}
@end

File: bus/BusViewModel.swift
//
//  BusViewModel.swift
//  bus
//
//  Created by liuhongnian on 2020/8/12.
//  Copyright © 2020 liuhongnian. All rights reserved.
//

import UIKit

class CZBusViewModel: NSObject {

    @objc dynamic var result: Dictionary<AnyHashable, Any>
    @objc var error: NSError?
    
    override init() {
        result = [:]
        error = nil
    }
    
}

class Bus59ViewModel: CZBusViewModel {
    
    private let busService = CZBusService.init()
    
    @objc func startRequest(withDirection direction: String, station: String) {
        

        busService.queryBus(roadName: "59", direction: direction, station: station) {[weak self] (busResult, error) in

            if (error == nil) {
                self?.result = busResult
            }else{
                self?.error = error
            }
        }
        
        
    }
    
}

class BRT1ViewModel: CZBusViewModel {
    
    private let busService = CZBusService.init()
    
    @objc func startRequest(withDirection direction: String, station: String) {
        
        busService.queryBus(roadName: "81", direction: direction, station: station) {[weak self] (busResult , error) in
            
            if (error == nil) {
                self?.result = busResult
            }else{
                self?.error = error
            }
        }
        
    }
}
